### ISSUE TITLE
getenv does not get environment var

### ISSUE THREAD

**Issue by claudiodaffra** [kripken/emscripten#3915]
Issue : getenv does not get Environment var
 
Excpet PATH that return '/' wrong path
the other environment var are NULL

#include <stdio.h>       
#include <stdlib.h>      

	int main ()
	{
	char* ENVAR;

		ENVAR = getenv ("PATH");		if (ENVAR!=NULL)	printf ("\n <%s>",ENVAR);	
		ENVAR = getenv ("TEMP");		if (ENVAR!=NULL)	printf ("\n <%s>",ENVAR);	
		ENVAR = getenv ("EMSCRIPTEN");	if (ENVAR!=NULL)	printf ("\n <%s>",ENVAR);	
		ENVAR = getenv ("ComSpec");		if (ENVAR!=NULL)	printf ("\n <%s>",ENVAR);	

	  return 0;
	}

**Comment by kripken** [comment]
Emscripten runs the code in a sandboxed environment, without direct access to the underlying platform. That's why we have a virtual filesystem, and we also have a virtual env var table, etc. I suppose you are talking here about node, which I guess does provide APIs to access the environment? If so we could add an option to access that.

**Comment by claudiodaffra** [comment]
yes in node is easy to get ENV VAR, i write my workaround ...

	EM_ASM ( "console.log('Current directory: ' + process.cwd());" );
	
thank you, kripken for your work about emscripten

**Comment by kripken** [comment]
Ok, renamed and marked as a good first bug. It should be fairly simple to add this support, for someone familiar with node, I would guess.

**Comment by claudiodaffra** [comment]
can you indicate me where  'getenv' is implemented ( which library ... js)
i send you somes pull request to implement getenv in node.js

**Comment by kripken** [comment]
Sure, look in `src/library.js` for `getenv`.

**Comment by claudiodaffra** [comment]
i have submitted a propose to change on emscriten.h 
to implement this macro emulating 'getenv'
i think is the easy way, as you can see
from under example.

the library.js ways is a bit more slow
to implement, because javascript
can't get the "name" of var
so when i pass var to getenv function
node.js get a bit confuse

decide you whart is better 

	#include <stdio.h>       
	#include <emscripten.h>

	using namespace std ;

	#define getenv(name)\
		(char*) EM_ASM_INT ( {\
		var envar = JSON.stringify(name);\
		var ret  =allocate(intArrayFromString(envar), 'i8', ALLOC_NORMAL);\
		return ret ;\
		}, NULL  ) 

		
	int main ()
	{
	 char *s = getenv ( process.env.APPDATA );
	 printf ( "\n NODE.JS <%s> " ,s  ) ; 
	 printf ( "\n NODE.JS <%s> " , getenv ( process.env.HOMEDRIVE ) ) ;
	 printf ( "\n NODE.JS <%s> " , getenv ( process.env ) ) ; 
	  
	  return 0;
	}
	
	
output


	 NODE.JS <"C:\\Users\\claudio\\AppData\\Roaming"> 
	 NODE.JS <"C:"> 
	 NODE.JS <{"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\claudio\\AppData\\Roaming","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","COMPUTERNAME":"DAFFRAA8","ComSpec":"C:\\WINDOWS\\system32\\cmd.exe","EMSCRIPTEN":"C:\\Program Files\\Emscripten\\emscripten\\1.35.0","EM_CONFIG":"C:\\Users\\claudio\\.emscripten","FPS_BROWSER_APP_PROFILE_STRING":"Internet Explorer","FPS_BROWSER_USER_PROFILE_STRING":"Default","FP_NO_HOST_CHECK":"NO","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\claudio","JAVA_HOME":"C:\\Program Files\\Emscripten\\java\\7.45_64bit","LOCALAPPDATA":"C:\\Users\\claudio\\AppData\\Local","LOGONSERVER":"\\\\DAFFRAA8","NUMBER_OF_PROCESSORS":"4","OS":"Windows_NT","Path":"C:\\Program Files\\Emscripten;C:\\Program Files\\Emscripten\\clang\\e1.35.0_64bit;C:\\Program Files\\Emscripten\\node\\4.1.1_64bit\\bin;C:\\Program Files\\Emscripten\\python\\2.7.5.3_64bit;C:\\Program Files\\Emscripten\\java\\7.45_64bit\\bin;C:\\Program Files\\Emscripten\\spidermonkey\\37.0.1_64bit;C:\\Program Files\\Emscripten\\emscripten\\1.35.0;C:\\Program Files\\Emscripten\\crunch\\1.03;C:\\Program Files\\Broadcom\\Broadcom 802.11;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\system32\\config\\systemprofile\\.dnx\\bin;C:\\Program Files\\Microsoft DNX\\Dnvm\\;C:\\Program Files (x86)\\GnuWin32\\bin;C:\\Program Files (x86)\\ATI Technologies\\ATI.ACE\\Core-Static;C:\\TDM-GCC-64\\bin","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"AMD64 Family 22 Model 48 Stepping 1, AuthenticAMD","PROCESSOR_LEVEL":"22","PROCESSOR_REVISION":"3001","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","PROMPT":"$P$G","PSModulePath":"C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules\\","PUBLIC":"C:\\Users\\Public","SESSIONNAME":"Console","SystemDrive":"C:","SystemRoot":"C:\\WINDOWS","TEMP":"C:\\Users\\claudio\\AppData\\Local\\Temp","TMP":"C:\\Users\\claudio\\AppData\\Local\\Temp","USERDOMAIN":"daffraa8","USERDOMAIN_ROAMINGPROFILE":"daffraa8","USERNAME":"claudio","USERPROFILE":"C:\\Users\\claudio","VS140COMNTOOLS":"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\Tools\\","windir":"C:\\WINDOWS"}>

**Comment by kripken** [comment]
Looks like that pull was closed, was that intentional?

**Comment by claudiodaffra** [comment]
yes beacuase there was a couple of programming errors, (memory leaks)
i have done ! ;-(
i'm testing new version 
and implemented now signal.h 
but i prefer test more deeply, before send you new code !
(without errors !) !

**Comment by claudiodaffra** [comment]
https://github.com/kripken/emscripten/pull/3948

**Comment by sbc100** [comment]
It seems that this can be done fairly easily with a pre-js file: https://github.com/emscripten-core/emscripten/pull/3948#issuecomment-744032264
